app/controllers/api/v1/topic_controller.rb

```diff
@@ -0,0 +1,12 @@
+# Class is responsible for handling the requests for the topics
+module Api
+  module V1
+    class TopicController < ApplicationController
+      before_action :authenticate_user!
+      include ActiveStorage::SetCurrent
```

Comment:
Probably you create an ApiController inside `api/v1`, move these lines there, and make this controller heritage from that new controller.

This way, all the controller inside api/v1 will have the before_action and the ActiveStorage configuration needed. Also, in the future you can rescue from some error there